applications/RomApplication/python_scripts/rom_nn_trainer.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import os
+import json
+
+import tensorflow as tf
+from keras.models import Model
+from keras import layers
+from keras.optimizers import AdamW
+from keras.callbacks import LearningRateScheduler
+
+import KratosMultiphysics
+
+
+class Rom_NN_trainer(object):
+
+    def __init__(self, general_rom_manager_parameters):
+
+        self.general_rom_manager_parameters = general_rom_manager_parameters
+        self.nn_parameters = self._SetNNParameters()
+
+    def _SetNNParameters(self):
+        defaults = self._GetDefaultNNParameters()
+
+        nn_params = self.general_rom_manager_parameters["NN"]
+
+        keys_to_copy = [
+             "saved_models_root_path"
+        ]
+
+        keys_to_copy_train = [
+             "modes",
+             "layers_size",
+             "batch_size",
+             "lr_strategy",
+             "epochs",
+             "database"
+        ]
+
+        keys_to_copy_online = [
+             "model_name"
+        ]
+
+        for key in keys_to_copy:
+            if key in nn_params.keys():
+                defaults[key] = nn_params[key]
+
+        if "training" in nn_params.keys():
+            for key in keys_to_copy_train:
+                if key in nn_params["training"].keys():
+                    defaults["training"][key] = nn_params["training"][key]
+
+        if "online" in nn_params.keys():
+            for key in keys_to_copy_online:
+                if key in nn_params["online"].keys():
+                    defaults["online"][key] = nn_params["online"][key]
+
```

Comment:
I don't understand what you're doing in here. Why don't creating the NN defaults as a standard Kratos parameters objects as we always do?

Reply:
Actually, I didn't know that there was already a method defined in the parameters class itself, to do the check. I'm removing this and using that one. Thanks for the comment!